Source/UI/NewUI/PaymentForm/ViewController/PaymentViewController.swift

```diff
@@ -319,25 +378,21 @@ extension PaymentViewController: BillingFormSummaryViewDelegate {
 }
 
 extension PaymentViewController: ExpiryDateViewDelegate {
-  func update(expiryDate: ExpiryDate) {
-    delegate?.expiryDateIsUpdated(value: expiryDate)
+  func update(result: Result<ExpiryDate, ExpiryDateError>) {
+    delegate?.expiryDateIsUpdated(result: result)
```

Comment:
Sorry @ehab-al-cko , I'll need to pull the plug again as I just came across this serious omission which we absolutely need to rectify.

In current design, our ViewController owns a ViewModel, but also owns a delegate, which is the ViewModel.

I will not block this PR for it, but please open a task and link it here, we'll need to cover that ASAP. To me right now it just looks like the **delegate should not exist**, given that delegate is the child view model already. It just doesn't make sense how we got here?

Reply:
It was before, if you want to refactoring then please add jira ticket.

Reply:
I think you have created this pattern and then built on it, from my side I've joined late and only just realised. If you are unhappy to review this behaviour and create a task for fixing it, I can do so from my side, sure.